test-integ/catalogv2/z_test.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package catalogv2
+
+import (
+	"flag"
+
+	"github.com/hashicorp/consul/test/integration/consul-container/libs/utils"
+	"github.com/hashicorp/consul/testing/deployer/topology"
+)
+
+var (
+	dev1_17Images = topology.Images{
+		ConsulCE:         "hashicorppreview/consul:1.17-dev",
```

Comment:
can we load these through defaulted envvars?

Reply:
yeah, what's the long term plan here? we're going to have to bump the version at some point I guess?

Reply:
Well ideally the `make test-compat-integ-setup` gets run, and then the images are the fresh ones from the dev build courtesy of `https://github.com/hashicorp/consul/blob/main/test/integration/consul-container/libs/utils/version.go#L59`

I was having a hard time producing a dev build of the `consul-dataplane` so I was just forcing all three here while I iterated.

I think one of the other CI jobs has some way to pin the version using an env var instead IIRC.

Reply:
I've updated this to:
- accept an env var from the GHA to force the dataplane version
- when not running in CI, substitute the 1.17 dev versions of images so you don't have to set the env var every time